pkg/inject/inject.go

```diff
@@ -362,10 +355,20 @@ func (conf *ResourceConfig) complete(template *v1.PodTemplateSpec) {
 	conf.pod.Meta = &template.ObjectMeta
 }
 
+// GetPod returns the parsed object as a *v1.Pod variable
+func (conf *ResourceConfig) GetPod() (*v1.Pod, error) {
```

Comment:
Probably not needed if you moved the logic of `webhook.getLabelForParent()` into this package.

Reply:
`inject.go` is not dependent on `k8s.API`, which is needed in `getLabelForParent()`. That's why I kept the pod object retrieval logic separated from the logic that retrieves its ownership, which is only used in the webhook.